src/ModelObjects/component.rs

```diff
@@ -91,18 +93,18 @@ impl Component {
         &mut self.declarations
     }
 
-    pub fn get_input_edges(&self) -> &Vec<Edge> {
+    pub fn get_input_edges(&self) -> Result<&Vec<Edge>, Box<dyn Error>> {
         if let Some(input_edges) = &self.input_edges {
-            input_edges
+            Ok(input_edges)
         } else {
-            panic!("attempted to get input edges before they were created")
+            bail!("attempted to get input edges before they were created")
         }
     }
-    pub fn get_output_edges(&self) -> &Vec<Edge> {
+    pub fn get_output_edges(&self) -> Result<&Vec<Edge>, Box<dyn Error>> {
         if let Some(output_edges) = &self.output_edges {
-            output_edges
+            Ok(output_edges)
         } else {
-            panic!("attempted to get output edges before they were created")
+            bail!("attempted to get output edges before they were created")
         }
     }
```

Comment:
It might be problematic having so 'common' methods return results, given that it is a very unexpected that it has not been initialized. It essentially means all functions calling it need to return results as well and so on recursively. Where do we draw the line between using panic! and bail!. This minor change for example means that every method accessing the inputs of a component must return a result. My thoughts are that having most functions return results is going to be slightly tedious, but I'm open for a discussion.